check50.py

```diff
@@ -553,6 +552,10 @@ def wait(self, timeout=1):
 
         self.output = "".join(self.output).replace("\r\n", "\n").lstrip("\n")
         self.kill()
+
+        if self.child.signalstatus == signal.SIGSEGV:
+            raise Error("failed to execute program due to segmentation fault")
```

Comment:
Hm, technically executed, it just crashed? Perhaps more simply "segmentation fault while running program"? Though do we have access to program name in this context to be more explicit?